test_runner/random_test_runner/Readme.md

```diff
@@ -0,0 +1,213 @@
+# Random test runner
+
+Random test runner allows running randomly generated scenarios to test autoware autonomy implementation.
+
+| NOTE: Currently only `autoware.auto` (`awf/auto`) is supported. `AutowareArchitectureProposal` (`tier4/proposal`) will be supported in the future. |
+|----------------------------------------------------------------------------------------------------------------------------------------------------|
+
+## How to build
+
+### Autoware.auto
+
+It is assumed that you are running AutowareAuto in ADE environment with default paths. See [Autoware.Auto documentation](https://autowarefoundation.gitlab.io/autoware.auto/AutowareAuto/installation-ade.html) for instructions how to correctly set up Autoware.Auto and ADE.
+
+Inside Autoware.auto directory run/enter ade:
+```shell
+ade --rc .aderc-amd64-foxy-lgsvl start --update --enter
+```
+
+enter autoware directory, install required dependencies and build
+
+```shell
+cd <autoware_auto_directory_name>
+vcs import < autoware.auto.foxy.repos
+./tools/simulation/get_scenario_simulator_v2.sh
+colcon build --packages-up-to random_test_runner scenario_simulator_launch kashiwanoha_map --cmake-args -DCMAKE_BUILD_TYPE=Release
+source install/setup.bash
+```
+
+### AutowareArchitectureProposal
+
+| NOTE: AutowareArchitectureProposal is not yet supported by random testing. |
+|----------------------------------------------------------------------------|
+
+## How to run
+
+First complete build described in [How to build](#how-to-build) section for autoware architecture you are using.
+
+Then, in first terminal run: 
+
+```shell
+rviz2 -d $(ros2 pkg prefix random_test_runner)/share/random_test_runner/rviz/random_test.rviz
+```
+
+In second:
+
+```shell
+ros2 launch random_test_runner random_test.launch.py
+```
+
+Which will run random tests with default paramters. You should see several npcs spawned in random locations around the ego vehicle, which will move on random path.
+
+After test is completed see `/tmp` directory. Among others, there will be two files:
+1. `result.junit.xml` - test result file with information about encountered errors
+2. `result.yaml` - yaml file that can be used to replay tests
+
+Exemplary content of those files can be seen below:
+
+`result.junit.xml`:
+```xml
+<?xml version="1.0"?>
+<testsuites failures="0" errors="6" tests="6">
+  <testsuite name="random_test14699091947383883875" failures="0" errors="6" tests="6">
```

Comment:
This data seems incorrect:
* `tests="6"` while there are only 5 tests;
* `errors="6"` while 4 out of 5 tests resulted in an error.

When I run it locally I have slightly different results:
```bash
<?xml version="1.0"?>
<testsuites failures="0" errors="4" tests="4">
  <testsuite name="random_test_12250249784762836567" failures="0" errors="4" tests="4">
    <testcase name="4">
      <error type="collision" message="npc9 and ego collided at 16.300000000000097s" />
      <error type="timeout" message="Ego failed to reach goal within timeout" />
    </testcase>
    <testcase name="2" />
    <testcase name="3">
      <error type="timeout" message="Ego failed to reach goal within timeout" />
    </testcase>
    <testcase name="1" />
    <testcase name="0">
      <error type="stand still" message="Ego seems to be stuck" />
    </testcase>
  </testsuite>
</testsuites>
```
I have `tests="4"` while 5 tests have been executed.

I am not sure if that metadata is even relevant...